src/http-util/src/lib.rs

```diff
@@ -105,3 +107,28 @@ pub async fn handle_prometheus(registry: &MetricsRegistry) -> impl IntoResponse
         .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?;
     Ok::<_, (StatusCode, String)>((TypedHeader(ContentType::text()), buffer))
 }
+
+/// Serves metrics from the selected metrics registry variant.
+#[allow(clippy::unused_async)]
+pub async fn handle_enable_otel(
+    callback: mz_ore::tracing::OpenTelemetryEnableCallback,
+    Json(payload): Json<Value>,
```

Comment:
I think just `Json<bool>` here will let you skip the manual schema validation:

```suggestion
    Json(payload): Json<bool>,
```

Though I'd advocate for:

```suggestion
#[derive(Serialize, Deserialize)]
struct DynamicOtelConfig {
    enabled: bool,
}

Json(payload): Json<DynamicOtelConfig>
```

so we can add more fields in the future.